lapis-docs/src/content/docs/concepts/advanced-query.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: Advanced query
+description: Generalized advanced queries
+---
+
+import { OnlyIf } from '../../../components/OnlyIf.tsx';
+import { hasFeature } from '../../../config.ts';
+
+{/* prettier-ignore */}
+<OnlyIf condition={hasFeature('generalizedAdvancedQuery')}>
+:::note
+This feature is not enabled for this LAPIS instance.
+It must be enabled in the `features` section of the database configuration by the maintainer of the instance.
+:::
+</OnlyIf>
+
+For those readers familiar with cov-spectrum's [variant query](../concepts/variant-query) feature advanced queries can be
+seen a superset of variant queries, in which not only variant information can be queried but also metadata information.
+
+Advanced queries can be used to filter sequences and metadata and can be passed to the server through the query parameter
+`advancedQuery`. Don't forget to encode/escape the query correctly (in JavaScript, this can be done with the
+[encodeURIComponent()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)
+function)!
+
+The formal specification of the query language is available
+[here](https://github.com/GenSpectrum/LAPIS/blob/main/lapis/src/main/antlr/org/genspectrum/lapis/model/advancedqueryparser/AdvancedQuery.g4)
+as an ANTLR v4 grammar.
+In following, we provide an informal description and examples.
+The respective
+[unit test](https://github.com/GenSpectrum/LAPIS/blob/main/lapis/src/test/kotlin/org/genspectrum/lapis/model/AdvancedQueryFacadeTest.kt)
+provides a full list of possible atomic queries.
+
+The query language understands Boolean logic. Expressions can be connected with `&` (and), `|` (or) and `!` (not).
+Both `&` and `AND` are recognized as `and`, `|` and `OR` are recognized as `or`, and `!` and `NOT` are recognized as `not`.
+Parentheses `(` and `)` can be used to define the order of the operations. Further, there is a special syntax to match
+sequences for which at least or exactly `n` out of a list of expressions are fulfilled. For variants we additionally add
+the `MAYBE` operator to query [ambiguous nucleotide symbols](../concepts/ambiguous-symbols).
+
+For metadata fields we also add custom queries to allow for regex substring search, is null queries (to search for fields
+which are empty or `null`) as well as range queries for dates and numbers.
+
+## Examples
+
+Get the sequences with the nucleotide mutation 300G, without a deletion at position 400 and either the AA change S:123T
+or the AA change S:234A:
+
+```
+300G & !400- & (S:123T | S:234A)
```

Comment:
Thanks very much for this PR, @anna-parker! Do we actually still want to support `&`, `|` and `!` or should we only keep `AND`, `OR` and `NOT`? Having both (especially when they are mixed) could make the queries more difficult to read for others and other programming languages usually also only support one syntax for this.

Reply:
I think it would be nice to ensure backward compatibility with covspectrum but removing  `&`, `|` and `!` would be fast if you think that is best

Reply:
(IMO this is a new language and we don't need to care about backwards compatibility too much. But that is just a slight preference.)

Reply:
As discussed we will keep the current state :-)